pkg/persistence/postgrespersister.go

```diff
@@ -171,62 +177,47 @@ func (p *PostgresPersister) retrieveDbEvents(tableName string, criteria *model.R
 	if criteria.Count > 0 {
 		queryBuf.WriteString(" LIMIT :count") // nolint: gosec
 	}
-
-	dbEvent := []postgres.Event{}
-	nstmt, err := p.db.PrepareNamed(queryBuf.String())
-	if err != nil {
-		return nil, err
-	}
-	err = nstmt.Select(&dbEvent, criteria)
-	return dbEvent, err
+	return queryBuf.String()
 }
 
-// PopulateBlockDataFromDB will fill the persistence with data from the DB.
-func (p *PostgresPersister) PopulateBlockDataFromDB(tableName string) error {
-	events, err := p.getLatestEvents(tableName)
+// actually can make this even more generic and put in utils
+func (p *PostgresPersister) saveEventToTable(query string, event *model.Event) error {
+	dbEvent, err := postgres.NewDbEventFromEvent(event)
 	if err != nil {
 		return err
 	}
-	// TODO: if you have 2 events w the same timestamp, just take the max block number
-	for _, event := range events {
-		modelEvent, err := event.DBToEventData()
-		if err != nil {
-			return err
-		}
-		blockData := PersisterBlockData{}
-		logPayload := modelEvent.LogPayload()
-		blockData.BlockNumber = logPayload.BlockNumber
-		blockData.BlockHash = logPayload.BlockHash
-		contractAddress := modelEvent.ContractAddress()
-		p.eventToLastBlockNumber[contractAddress] = make(map[string]PersisterBlockData)
-		p.eventToLastBlockNumber[contractAddress][modelEvent.EventType()] = blockData
+	_, err = p.db.NamedExec(query, dbEvent)
+	if err != nil {
+		return fmt.Errorf("Error saving event to table %v", err)
 	}
 	return nil
 }
 
 func (p *PostgresPersister) getLatestEvents(tableName string) ([]postgres.Event, error) {
-	query := p.retrieveLatestEventsQueryString(tableName)
+	queryString := p.retrieveLatestEventsQueryString(tableName)
 	events := []postgres.Event{}
-	err := p.db.Select(&events, query)
+	err := p.db.Select(&events, queryString)
 	return events, err
 }
 
+// retrieveLatestEventsQueryString queries for the events with the latest timestamp given an event type and contract address
 func (p *PostgresPersister) retrieveLatestEventsQueryString(tableName string) string {
 	// Query for the latest timestamp.
-	return fmt.Sprintf("SELECT e.event_type, e.log_payload, e.payload, e.hash, e.contract_address, e.contract_name, max_e.timestamp "+
-		"FROM (SELECT event_type, contract_address, MAX(timestamp) AS timestamp FROM %s GROUP BY event_type, contract_address) max_e "+
-		"JOIN %s e ON e.event_type = max_e.event_type AND e.timestamp = max_e.timestamp AND e.contract_address = max_e.contract_address;", tableName, tableName)
+	return fmt.Sprintf(`SELECT e.event_type, e.log_payload, e.payload, e.hash, e.contract_address, e.contract_name, max_e.timestamp
```

Comment:
Minor and up to you, but I wonder if you can use `StructFieldsForQuery` here for the field names.  Can modify to include a prefix value for each field.

Reply:
yeah that makes sense